app/code/Magento/Rss/Model/Rss.php

```diff
@@ -59,54 +62,56 @@ public function __construct(
     }
 
     /**
+     * Returns feeds
+     *
      * @return array
      */
     public function getFeeds()
     {
         if ($this->dataProvider === null) {
             return [];
         }
-        $cache = false;
-        if ($this->dataProvider->getCacheKey() && $this->dataProvider->getCacheLifetime()) {
-            $cache = $this->cache->load($this->dataProvider->getCacheKey());
-        }
+        $cacheKey = $this->dataProvider->getCacheKey();
+        $cacheLifeTime = $this->dataProvider->getCacheLifetime();
 
+        $cache = $cacheKey && $cacheLifeTime ? $this->cache->load($cacheKey) : false;
         if ($cache) {
             return $this->serializer->unserialize($cache);
         }
 
-        $data = $this->dataProvider->getRssData();
+        $serializedData = $this->serializer->serialize($this->dataProvider->getRssData());
 
-        if ($this->dataProvider->getCacheKey() && $this->dataProvider->getCacheLifetime()) {
-            $this->cache->save(
-                $this->serializer->serialize($data),
-                $this->dataProvider->getCacheKey(),
-                ['rss'],
-                $this->dataProvider->getCacheLifetime()
-            );
+        if ($cacheKey && $cacheLifeTime) {
+            $this->cache->save($serializedData, $cacheKey, ['rss'], $cacheLifeTime);
         }
 
-        return $data;
+        return $this->serializer->unserialize($serializedData);
```

Comment:
why do we need to serialize it and unserialize again?

Reply:
To be sure that it returns an array that will not consist Phrase object. And also it is one place to fix NewProducts, Wishlist, Product Special, etc.

Reply:
Ok, it’s not obvious, in this case would be good to add comment